proto/config/netcmn.proto

```diff
@@ -114,3 +114,29 @@ enum WiFiKeyScheme {
   WPAPSK = 1;        // WPA-PSK
   WPAEAP = 2;        // WPA-EAP or WPA2 Enterprise
 }
+
+// User-defined method to use to determine the connectivity status.
+enum ConnectivityProbeMethod {
+  // Connectivity probing method is not specified and therefore will not be used.
+  CONNECTIVITY_PROBE_METHOD_UNSPECIFIED = 0;
+  // Use ICMP ping against the probed endpoint to determine the connectivity status.
+  CONNECTIVITY_PROBE_METHOD_ICMP = 1;
+  // Try to establish TCP connection with the probed endpoint to determine the connectivity
+  // status.
+  CONNECTIVITY_PROBE_METHOD_TCP = 2;
+  // TODO: Consider adding support for BFD probing.
+}
+
+// Configuration for user-defined connectivity-testing probe.
+message ConnectivityProbe {
+  // Method to use to determine the connectivity status.
+  ConnectivityProbeMethod probe_method = 1;
+  // IP or FQDN to probe using the selected probing mechanism to determine
+  // the connectivity status.
+  // For CONNECTIVITY_PROBE_METHOD_TCP the address should include port number
+  // in one of the formats:
+  //   ipv4:port
+  //   [ipv6]:port
+  //   hostname:port
+  string probe_address = 2;
```

Comment:
what about 'probe_endpoint' which is address plus port at least for TCP type

Reply:
Done, added `message ProbeEndpoint`.